analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -44,10 +44,7 @@ private static ProgramState LearnBranchingConstraint<T>(ProgramState state, IBin
             if ((OperandSymbolWithoutConstraint(binary.LeftOperand) ?? OperandSymbolWithoutConstraint(binary.RightOperand)) is { } testedSymbol
                 && (OperandConstraint(binary.LeftOperand) ?? OperandConstraint(binary.RightOperand)) is { } constraint)
             {
-                if (useOpposite ^ binary.OperatorKind is BinaryOperatorKind.NotEquals or BinaryOperatorKind.ObjectValueNotEquals)
-                {
-                    constraint = constraint.Opposite;
-                }
+                constraint = constraint.ApplyOpposite(useOpposite ^ binary.OperatorKind is BinaryOperatorKind.NotEquals or BinaryOperatorKind.ObjectValueNotEquals);
```

Comment:
Should use the new extension method.

Reply:
It was done in the refactoring when introducing the method. So rebase fixes it.